File: compiler/src/main/java/com/aaroncoplan/waterfall/compiler/statements/ArrayLiteralData.java
package com.aaroncoplan.waterfall.compiler.statements;

import com.aaroncoplan.waterfall.generated.WaterfallParser;

import java.util.List;
import java.util.stream.Collectors;

public class ArrayLiteralData {

    public final List<ExpressionData> elements;

    public ArrayLiteralData(String filePath, WaterfallParser.ArrayLiteralContext ctx) {
        this.elements = ctx.positionalArgumentList().expression().stream()
                .map(e -> new ExpressionData(filePath, e))
                .collect(Collectors.toList());
    }

}

File: compiler/src/main/kotlin/com/aaroncoplan/waterfall/compiler/statements/ArrayLiteralData.kt
package com.aaroncoplan.waterfall.compiler.statements

import com.aaroncoplan.waterfall.generated.WaterfallParser

class ArrayLiteralData(filePath: String, ctx: WaterfallParser.ArrayLiteralContext) {
    @JvmField val elements: List<ExpressionData> =
        ctx.positionalArgumentList().expression().map { ExpressionData(filePath, it) }
}
